x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/rules/utils.test.ts

```diff
@@ -20,535 +20,105 @@ import {
-import { getSimpleRule } from '../__mocks__/utils';
+import { getSimpleRule, getOutputRuleAlertForRest } from '../__mocks__/utils';
 import { createRulesStreamFromNdJson } from '../../rules/create_rules_stream_from_ndjson';
 import { createPromiseFromStreams } from '../../../../../../../../../src/legacy/utils/streams';
 import { PartialAlert } from '../../../../../../../../plugins/alerting/server';
 import { SanitizedAlert } from '../../../../../../../../plugins/alerting/server/types';
+import { RuleAlertType } from '../../rules/types';
+import { setFeatureFlagsForTestsOnly, unSetFeatureFlagsForTestsOnly } from '../../feature_flags';
 
 type PromiseFromStreams = ImportRuleAlertRest | Error;
 
 describe('utils', () => {
+  beforeAll(() => {
+    setFeatureFlagsForTestsOnly();
+  });
+
+  afterAll(() => {
+    unSetFeatureFlagsForTestsOnly();
+  });
+
   describe('transformAlertToRule', () => {
     test('should work with a full data set', () => {
       const fullRule = getResult();
       const rule = transformAlertToRule(fullRule);
-      const expected: OutputRuleAlertRest = {
-        created_by: 'elastic',
-        created_at: '2019-12-13T16:40:33.400Z',
-        updated_at: '2019-12-13T16:40:33.400Z',
-        description: 'Detecting root and admin users',
-        enabled: true,
-        false_positives: [],
-        from: 'now-6m',
-        id: '04128c15-0d1b-4716-a4c5-46997ac7f3bd',
-        immutable: false,
-        index: ['auditbeat-*', 'filebeat-*', 'packetbeat-*', 'winlogbeat-*'],
-        interval: '5m',
-        risk_score: 50,
-        rule_id: 'rule-1',
-        language: 'kuery',
-        max_signals: 100,
-        name: 'Detect Root/Admin Users',
-        output_index: '.siem-signals',
-        query: 'user.name: root or user.name: admin',
-        references: ['http://www.example.com', 'https://ww.example.com'],
-        severity: 'high',
-        updated_by: 'elastic',
-        tags: [],
-        threat: [
-          {
-            framework: 'MITRE ATT&CK',
-            tactic: {
-              id: 'TA0040',
-              name: 'impact',
-              reference: 'https://attack.mitre.org/tactics/TA0040/',
-            },
-            technique: [
-              {
-                id: 'T1499',
-                name: 'endpoint denial of service',
-                reference: 'https://attack.mitre.org/techniques/T1499/',
-              },
-            ],
-          },
-        ],
-        filters: [
-          {
-            query: {
-              match_phrase: {
-                'host.name': 'some-host',
-              },
-            },
-          },
-        ],
-        meta: {
-          someMeta: 'someField',
-        },
-        timeline_id: 'some-timeline-id',
-        timeline_title: 'some-timeline-title',
-        to: 'now',
-        type: 'query',
-        note: '# Investigative notes',
-        version: 1,
-      };
-      expect(rule).toEqual(expected);
+      expect(rule).toEqual(getOutputRuleAlertForRest());
```

Comment:
Thanks for moving that out, makes the tests so much easier to read.

Reply:
Yeah I needed to cut down on boiler plate and "touch points" for when we create data structures in the REST code and tests.